index.js

```diff
@@ -24,70 +46,79 @@ function getFileExtension(URL){
-//     // /* Match only links that are fully qualified with https */
-//     //const fullLinkOnlyRegex = /^\[([\w\s\d]+)\]\((https?:\/\/[\w\d./?=#]+)\)$/
-//     // /* Match full links and relative paths */
-//     //const regEx = /^\[([\w\s\d]+)\]\(((?:\/|https?:\/\/)[\w\d./?=#]+)\)$/
-//     const myMatch = data.replace(/[{()}]/g, ' ').match(regexp)
-//     console.log(myMatch)
-//     return myMatch;
-//     /* ["[View the analytics docs](https://getanalytics.io/)", "View the analytics docs", "https://getanalytics.io/", index: 0, input: "[View the analytics docs](https://getanalytics.io/)", groups: undefined] */
-
-//     // // de-structure the array
-//     // const [ full, text, url ] = myMatch
-
-//     // console.log(text)
-//     // // 'View the analytics docs'
-
-//     // console.log(url)
-//     // // 'https://getanalytics.io/'
-//   }
-//   getLinksInFiles('./README.md');
-// //  console.log('PRINTING FILE CONTENT FROM', readFiles('./README.md'))
-// // // // console.log('PRINTING FILE EXTENSION FROM', getFileExtension('./test.md'))
-// //  console.log('GETING LINKS IN A FILE', getLinksInFiles('./README.md'))
 
+//------------EXERCISE OF RECURSIVNESS WITH klawSync-------
+const mdFilesInDir = (path) => {
+    const filesInDir = klawSync(path) //Return an array of objetcs with all the files that exist inside ./sampleFiles directory.
+    //console.log(filesInDir);
+    let mdFiles = [];
+
+    filesInDir.forEach((file) => {
+      const filePath = file.path; // Getting just the path property of the arrOfFiles
+      //console.log(filePath);
+      const fileExtension = getFileExtension(filePath); // Getting the file´s extension
+      //console.log(fileExtension)
+        if(fileExtension === '.md'){
+          mdFiles.push(filePath);
+
+
+        //console.log(chalk.hex('#cbaacb')('List of .md files:'), chalk.hex('#6c88c4')(mdFiles));
+        // //console.log("List of .md files:" + chalk.pink + filePath);
+        // const linksInAFile = arrOfLinks(mdFiles);
+        // //console.log(linksInAFile);
+        // console.log(chalk.hex('ffffb5')("Links in the .md file:") , linksInAFile);
+        //   // const mdFiles = filesInDir.filter((extension) => {
+        //   //   if(fileExtension === '.md')
+        //   //   console.log(filesInDir);
+
+        //   // })
+        //console.log(mdFiles)
+        //     return mdFiles;
+       };
+
+    // });
+    // return filesInDir;
+  })
+    
+  console.log(mdFiles);
+    return mdFiles;
+}
+
+//mdFilesInDir('./sampleFiles');
+
+
+function mdLinks(path, options) {
+
+  const listOfMdFiles = mdFilesInDir(path);
+  // listOfMdFiles.forEach((file) => {
+  //console.log(chalk.hex('#cbaacb')('List of .md files:\n') + chalk.hex('#6c88c4')(listOfMdFiles));
+  // // // //console.log("List of .md files:" + chalk.pink + filePath);
+  listOfMdFiles.forEach(file => {
+    console.log(chalk.hex('#cbaacb')('List of .md files:\n') + chalk.hex('#6c88c4')(listOfMdFiles));
```

Comment:
el elemento largo que refieres es el `listOfMdFiles` - este es un array si? entonces el `console.log` esta intentando , o quiza con chalk afecta tambien, a imprimir este array.
entonces esta imprimiendo `listOfMdFiles` como un comma delimited string. Es mi hipotesis.
Fijate que en cada llamada de tu callback, `file` es el path md actualizado en este momento, osea `listOfMdFiles[i]`

Reply:
also a sidenote `forEach` callback has 3 params, the current element (`file` in this case) and two optional params, the current `index` (or loop iteration) and the full array.
so `forEach` for example can be 
```
array.forEach((element, i, arr) => console.log(`I am ${element} in position ${i} of this array ${arr}`);
```